cmd/report.go

```diff
@@ -10,28 +10,20 @@ import (
 
 	"go.k6.io/k6/execution"
 	"go.k6.io/k6/lib/consts"
+	"go.k6.io/k6/usage"
 )
 
-type report struct {
-	Version    string         `json:"k6_version"`
-	Executors  map[string]int `json:"executors"`
-	VUsMax     int64          `json:"vus_max"`
-	Iterations uint64         `json:"iterations"`
-	Duration   string         `json:"duration"`
-	GoOS       string         `json:"goos"`
-	GoArch     string         `json:"goarch"`
-	Modules    []string       `json:"modules"`
-	Outputs    []string       `json:"outputs"`
-}
-
-func createReport(execScheduler *execution.Scheduler, importedModules []string, outputs []string) report {
-	executors := make(map[string]int)
+func createReport(
+	u *usage.Usage, execScheduler *execution.Scheduler, importedModules []string, outputs []string,
+) (map[string]any, error) {
 	for _, ec := range execScheduler.GetExecutorConfigs() {
-		executors[ec.GetType()]++
+		err := u.Uint64("executors/"+ec.GetType(), 1)
+		if err != nil { // TODO report it as an error but still send the report?
```

Comment:
Couldn't we `panic` here? 🤔 

As I stated before, I think it's better to make `Uint64` to return the error, because it's like a library, and especially if we have the idea of exposing that report/usage package to extensions.

But, from the consumer perspective, at this point, this should technically never fail, right?

Reply:
I really prefer that k6 never panics - especially not on usage reporting.

> But, from the consumer perspective, at this point, this should technically never fail, right?

There is none zero chance someone will use this API and add a string with `exucutors/ramping-arrival-rate` - is this *likely*, no. I would argue it will likely be a try at breaking something. But I still don't think it is justified to panic. 

Panicking in this case just means that we will get a users test stopped, potentially with a lot of confusion.

I would prefer if this usage accumulation also happens outside of here for all cases - this likely won't be one of them.

I can remove the error handling as a whole for this case as well, so we again get a report even if this happens.

Reply:
Well, yeah... That's why I also suggested to have some reserved keywords - of course it should never panic, and even less for a non-critical feature like usage reporting, but having reports with basic information `k6_version` faked will also be confusing - as you said _will likely be a try at breaking something_.

Overall, and to be clear, my intention when suggesting to `panic` here, is to early capture any programming mistake, if the `panic` can be "forced" from "outside", then I agree is a no-go.

> I can remove the error handling as a whole for this case as well, so we again get a report even if this happens.

Perhaps a good idea, yeah. Although, to be honest, I don't feel really attracted by any approach. I wrote the comment because you left the TODO, but any approach that looks good for you will most likely do it for me as well 👍🏻

Reply:
> some reserved keywords - 

There in practice are - all of the ones we add directly to the map.

For me having reserved words is strange as ... well it just means we now centralize the usage reporting by having some names codefied in a central place ... 

I will drop the error reporting on this and a different TODO, will also try to move the reporting away from thsi function for the cases where this is possible.